src/test/scala/za/co/absa/commons/io/TempDirectorySpec.scala

```diff
@@ -45,23 +46,23 @@ class TempDirectorySpec extends AnyFlatSpec with Matchers {
     name3 should (startWith("foo") and endWith("bar"))
   }
 
-  behavior of "`toString`"
+  behavior of "`asString`"
 
-  it should "return valid string path" in {
-    val tempDirectory = TempDirectory().deleteOnExit()
-    val tempDirectoryPath: String = tempDirectory.toString
-    val expectedPath = tempDirectory.path.toAbsolutePath.toString.replace("\\", "/")
+  it should "return valid string path with unix slashes" in {
+    val tempDirectory = TempDirectory(prefix = "fake_tmp_").deleteOnExit()
 
-    tempDirectoryPath should equal(expectedPath)
+    tempDirectory.asString should include("/fake_tmp_")
   }
 
   behavior of "`toURI`"
 
   it should "return valid URI" in {
-    val tempDirectory = TempDirectory().deleteOnExit()
-    val expectedURIString = s"file:/${tempDirectory.toString}/".replace("//", "/")
+    val tempDirectory = TempDirectory(prefix = "fake_tmp_").deleteOnExit()
 
-    tempDirectory.toURI.toString should equal(expectedURIString)
+    tempDirectory.toURI shouldNot be(null)
```

Comment:
This is really minor, but if you would wish to avoid using `null` then you could operate around it using `Option`.
```suggestion
    Option(tempDirectory.toURI)  should not be empty
```
nebo
```suggestion
    Option(tempDirectory.toURI)  shouldNot be(empty)
```